### PR TITLE
Confine GRUB providers to presence of menus, prefer GRUB 2

### PR DESCRIPTION
**Pull request by ckoenig**
This pull request is based on https://github.com/hercules-team/augeasproviders_grub/pull/1 and the latest comment by @m4ce. The specs are still working.

### CODE REVIEW

File: lib/puppet/provider/kernel_parameter/grub.rb [hunk 1]
```diff
@@ -13,6 +13,7 @@
   lens { 'Grub.lns' }
 
   confine :feature => :augeas
+  confine :grubversion => 1
```
Comment (vinzent) [inline lib/puppet/provider/kernel_parameter/grub.rb]: why not use defaultfor? maybe there are people around that wan't to the old grub or the other way around. but if you confine the version you can't set the provider on the resource with "provider => grub2"

**Comment by ckoenig** [comment]
I added two commits incorporating your feedback. Thanks.

**Comment by raphink** [comment]
@domcleal what do you think of this PR?

**Comment by raphink** [comment]
@domcleal ?

**Comment by domcleal** [comment]
Seems reasonable I guess.

File: lib/facter/augeasprovider_grub_version.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+Facter.add(:augeasprovider_grub_version) do
+  if File.exists?('/etc/default/grub')
+    version = 2
+  elsif File.exists?('/boot/efi/EFI/redhat/grub.conf') || File.exists?('/boot/grub/menu.lst')
+    version = 1
+  else
+    fail('Failed to evaluate grub version')
```
Comment (raphink) [inline lib/facter/augeasprovider_grub_version.rb]: What happens in facter when the fact fails? Shouldn't it simply return `nil` instead?

Comment (trevor-vaughan) [inline lib/facter/augeasprovider_grub_version.rb]: Yes. Facts should never fail.

**Comment by ckoenig** [comment]
To be honest, I'm not sure, what to do about the failing builds in travis. Looks like it tries to install a wrong rake version. Should I rebase by branch on the current master or is the problem something else?

**Comment by bastelfreak** [comment]
Some of the gems do not work on ruby 1.8.3 in their latest version. there were some changes in the Gemfile that may get them working again.

Or would it be possible to get this merged without without passing tests on 1.8.3? Everything else is green. This would probably fix my issue in https://github.com/hercules-team/augeasproviders_grub/issues/22

**Comment by raphink** [comment]
https://github.com/hercules-team/augeasproviders_grub/commit/0d8f4455d4278b02abd2c5dca772eaf1fa637fe5 fixes build on Ruby 1.8, so a rebase should pass now.

**Pull request merged**